### DISCUSSION TITLE
Stub run and cpu/memory usage

### DISCUSSION THREAD

**Discussion by Mxrcon** [nextflow-io/nextflow#2094] [Ideas]
Hey there,
I'd like to discuss the usage of cpu and memory when you run a stub run. I was trying to debug my stubs and then got an error `Process requirement exceed available CPUs -- req: 8; avail: 2`, I'm using a very basic computer just for debugging propose and i think that a configuration that limits how many cpu stubs use could be a nice addition, because when you're using stubs, nextflow don't need a lot of cpus cores or Gbs of ram. 
I'm currently using a stub profile to match the ram of my computer, as a way of working around my computer limitations.

Regards, Davi

**Comment by pdp10** [comment]
I have the same problem.

From a Nextflow [commit](https://github.com/jordeu/nextflow/commit/440b15b17d9048da34aeaa2a85d9039bd736bb4d), I found the attribute `${workflow.stubRun}`, which works within a process.
I thought about using it in a conditional inside the config process block, but this does not work: 
```
process {
    if ( "${workflow.stubRun}" == "false" ) {
         // configure process with cpus/memory
   }
}
```

@pditommaso could you let us know whether this is possible or the best approach to address this, please?

**Reply by pditommaso** [reply]
You can create two different config profiles having different cpus/mem settings. For example: 

```
profiles { 
  production {
     process.memory = '8 GB' 
     process.cpus = 4
  }

  testing {
    stubRun = true
    process.memory = '1 GB' 
    process.cpus = 1    
  }
} 
```

**Reply by pdp10** [reply]
> testing {
>     stubRun = true
>     process.memory = '1 GB' 
>     process.cpus = 1    
>   }

Thank you for pointing me out to the right direction and with a very neat approach too.
